rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake

```diff
@@ -0,0 +1,288 @@
+
+set(_target_suffix "__rosidl_typesupport_opensplice_c")
+
+# avoid generating any opensplice specific stuff for builtin_interfaces
+if(NOT "${PROJECT_NAME} " STREQUAL "builtin_interfaces ")
+
+set(_dds_idl_files "")
+set(_dds_idl_base_path "${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_dds_idl")
+foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
+  get_filename_component(_extension "${_idl_file}" EXT)
+  get_filename_component(_name "${_idl_file}" NAME_WE)
+  if("${_extension} " STREQUAL ".msg ")
+    get_filename_component(_parent_folder "${_idl_file}" DIRECTORY)
+    get_filename_component(_parent_folder "${_parent_folder}" NAME)
+    list(APPEND _dds_idl_files
+      "${_dds_idl_base_path}/${PROJECT_NAME}/${_parent_folder}/dds_opensplice/${_name}_.idl")
+  elseif("${_extension} " STREQUAL ".srv ")
+    list(APPEND _dds_idl_files
+      "${_dds_idl_base_path}/${PROJECT_NAME}/srv/dds_opensplice/Sample_${_name}_Request_.idl")
+    list(APPEND _dds_idl_files
+      "${_dds_idl_base_path}/${PROJECT_NAME}/srv/dds_opensplice/Sample_${_name}_Response_.idl")
+  endif()
+endforeach()
+
+set(_output_path "${CMAKE_CURRENT_BINARY_DIR}/rosidl_typesupport_opensplice_c/${PROJECT_NAME}")
+set(_dds_output_path "${CMAKE_CURRENT_BINARY_DIR}/rosidl_typesupport_opensplice_cpp/${PROJECT_NAME}")
+set(_generated_msg_files "")
+set(_generated_external_msg_files "")
+set(_generated_srv_files "")
+set(_generated_external_srv_files "")
+foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
+  get_filename_component(_extension "${_idl_file}" EXT)
+  get_filename_component(_msg_name "${_idl_file}" NAME_WE)
+  string_camel_case_to_lower_case_underscore("${_msg_name}" _header_name)
+  if("${_extension} " STREQUAL ".msg ")
+    get_filename_component(_parent_folder "${_idl_file}" DIRECTORY)
+    get_filename_component(_parent_folder "${_parent_folder}" NAME)
+    list(APPEND _generated_external_msg_files
+      "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_msg_name}_.h"
+      "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_msg_name}_.cpp"
+      "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_msg_name}_Dcps.h"
+      "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_msg_name}_Dcps.cpp"
+      "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_msg_name}_Dcps_impl.h"
+      "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_msg_name}_Dcps_impl.cpp"
+      "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_msg_name}_SplDcps.h"
+      "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_msg_name}_SplDcps.cpp"
+      "${_dds_output_path}/${_parent_folder}/dds_opensplice/ccpp_${_msg_name}_.h")
+    list(APPEND _generated_msg_files
+      "${_output_path}/${_parent_folder}/dds_opensplice_c/${_header_name}__type_support_c.cpp")
+  elseif("${_extension} " STREQUAL ".srv ")
+    # list(APPEND _generated_srv_files "${_output_path}/srv/dds_opensplice_c/${_header_name}__type_support_c.cpp")
```

Comment:
Are you saving this comment for later (when you determine if generation for services is needed)?